File: task-03/Python/Sub-task 4/sub-task-4.py
with open('input.txt','r') as input:
	n=int(input.read())

with open('output.txt', 'w') as output:
    for i in range(n):
        for j in range(i, n):
            output.write("  ")
        for j in range(i + 1):
            output.write("* ")
        for j in range(i):
            output.write("* ")
        output.write("\n")
    for i in range(n + 1):
        for j in range(i):
            output.write("  ",)
        for j in range(i, n):
            output.write("* ")
        for j in range(i, n + 1):
            output.write("* ")
        output.write("\n")

File: task-03/Rust/Sub-task 4/sub-task-4.rs
use std::fs::File;
use std::io::{BufRead, BufReader, Write};

fn main() -> std::io::Result<()> {
    let input = File::open("input.txt")?;
    let mut reader = BufReader::new(input);
    let mut n_str = String::new();
    reader.read_line(&mut n_str)?;
    let n: usize = n_str.trim().parse().expect("Failed to parse input as integer");

    let mut output = File::create("output.txt")?;

    for i in 0..n {
        for _ in i..n {
            write!(output, "  ")?;
        }
        for _ in 0..=i {
            write!(output, "* ")?;
        }
        for _ in 0..i {
            write!(output, "* ")?;
        }
        writeln!(output)?;
    }

    for i in 0..=n {
        for _ in 0..i {
            write!(output, "  ")?;
        }
        for _ in i..n {
            write!(output, "* ")?;
        }
        for _ in i..=n {
            write!(output, "* ")?;
        }
        writeln!(output)?;
    }

    Ok(())
}
